dot/parachain/provisioner/messages/messages.go

```diff
@@ -0,0 +1,45 @@
+package provisionermessages
+
+import (
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+var (
+	_ Data = (*ProvisionableDataBackedCandidate)(nil)
+	_ Data = (*ProvisionableDataMisbehaviorReport)(nil)
+)
+
+type RequestInherentData struct {
+	RelayParent             common.Hash
+	ProvisionerInherentData chan ProvisionerInherentData
+}
+
+type ProvisionerInherentData struct {
+}
+
+// ProvisionerMessageProvisionableData is a provisioner message.
+// This data should become part of a relay chain block.
+type ProvisionableData struct {
+	RelayParent common.Hash
+	Data        Data
+}
+
+// Data becomes intrinsics or extrinsics which should be included in a future relay chain block.
+type Data interface {
+	IsProvisionableData()
```

Comment:
I would say that `IsProvisionableData` can be confused with the struct `ProvisionableData`, `isData` looks a better name to group everything that belongs to `Data` interface
```suggestion
	IsData()
```

Reply:
Made the asked changes! This was confusing since rust has a struct named `ProvisionableData` and then an enum named `ProvisionableData` inside of it

Reply:
https://github.com/paritytech/polkadot-sdk/blob/6c2b46f9620fd147e63b151320cc145551fd2c18/polkadot/node/subsystem-types/src/messages.rs#L843